pkg/cloud/azure/actuators/machine/reconciler.go

```diff
@@ -474,6 +477,15 @@ func (s *Reconciler) createVirtualMachine(ctx context.Context, nicName string) e
 			Zone:       vmZone,
 		}
 
+		userData, userDataErr := s.getCustomUserData()
```

Comment:
>// CustomData - Specifies a base-64 encoded string of custom data. The base-64 encoded string is decoded to a binary array that is saved as a file on the Virtual Machine. The maximum length of the binary array is 65535 bytes. <br><br> For using cloud-init for your VM, see [Using cloud-init to customize a Linux VM during creation](https://docs.microsoft.com/azure/virtual-machines/virtual-machines-linux-using-cloud-init?toc=%2fazure%2fvirtual-machines%2flinux%2ftoc.json)

This would need to go through `base64.StdEncoding.EncodeToString() to be consumable by `OSProfile.CustomData`

Reply:
Nice catch, i was referring to the parent PR and i missed porting it over, fixing it now and adding test to catch this